### PR TITLE
feat: add Neighborhood Suitability Map

### PR DESCRIPTION
**Pull request by stdavis**
@gregbunce This is an example of how your map could be embedded within a post. Not sure if you want to work off of this PR or copy the map code into your own.

A few things to note:
* I'm not sure why the color ramps seems to be applied differently in this map as compared to your original one
* I replaced the mapbox base map with an OSM one that doesn't require an API key. I didn't think that it was a great idea to post the key in GitHub. I realize that the map isn't as muted as your original one. We can investigate what we can do to help this including perhaps using discover?

Link to the example post: https://deploy-preview-1653--gis-utah-gov.netlify.app/neighborhood-level-suitability-analysis-in-utah/

### CODE REVIEW

**Comment by gregbunce** [comment]
Thank you, @stdavis!  you're amazing.  it looks great!  and i think the OSM basemap is fine.  I agree, i wasn't happy about putting my mapbox token out there.  also, i think the color ramp is fine.

a couple of questions

- is it possible to put the sliders on the left side of the map, in the same HTML block?  i'm not sure how that will work on mobile, but the idea being that the user can see the sliders and the map at the same time.  or, maybe it's better to have this as a web app - and i can link to it from the post.  what are your thoughts on of both ideas?
- can we have the sliders off (value = 0) on page load?  that might be better as the user can define their own suitability analysis. 
- it seems like the sliders need to be fully on (value = 1) or fully off (value = 0) in order for the map to update.  seems like the middle steps are temperamental.  are you seeing, too?

Again, maybe it's better suited as a separate web app (?).

Thank you again!  If I wasn't running behind schedule I'd attempt to tweak the code you created.

**Comment by stdavis** [comment]
> * is it possible to put the sliders on the left side of the map, in the same HTML block

Yes, I can do that.

> * or, maybe it's better to have this as a web app

Based on your comments from the chat about Matt wanting to use the map for other purposes, I'll move it to a separate page. @steveoh Do you have any issues with it being a page on the website, or should we stick it on mapserv?

> * can we have the sliders off (value = 0) on page load?

👍

> * it seems like the sliders need to be fully on (value = 1) or fully off (value = 0) in order for the map to update.  seems like the middle steps are temperamental.  are you seeing, too?

I'm _not_ seeing this on Chrome (macOS). Might be a browser thing. What browser and OS are you using?

**Comment by gregbunce** [comment]
Great, thanks again Scott.  

I'm using Chrome (windows).  It seems to be better now.  i just tried FireFox, too and it seems pretty responsive.

**Comment by steveoh** [comment]
> Based on your comments from the chat about Matt wanting to use the map for other purposes, I'll move it to a separate page. @steveoh Do you have any issues with it being a page on the website, or should we stick it on mapserv?

my opinion is that if it is to support a blog post jam it in the website. if it is going to be a more general resource, put it on mapserv

**Comment by gregbunce** [comment]
i'd say more of a general resource (that will also be referenced and used in a blog post).  WFRC and Human Services are interested in exploring the map a bit.  mapserv sounds like a good idea to me. scott also mentioned it can be embedded in the post.  so, that would be perfect!

**Comment by stdavis** [comment]
Hey @gregbunce,

This is ready for another look. I moved the map to its own page: https://deploy-preview-1653--gis-utah-gov.netlify.app/developer/applications/suitability and then included it in the [example blog post](https://deploy-preview-1653--gis-utah-gov.netlify.app/neighborhood-level-suitability-analysis-in-utah/) via an iframe.

**Comment by steveoh** [comment]
@stdavis if you are interested there is a `/scripts` folder and a grunt task to minify javascript. Just a heads up that it's set up if you want to use it.

**Comment by gregbunce** [comment]
this looks great!  the interface is much better.  i'd like to add a title, some text describing what it is, and then a "here's how to use this".  all really brief.  should i pass that text off to you or should i fork it, or make suggestions to this pr?  i guess we could also merge this pr (when stable) and i could work off of a new branch -working with this hidden page.  thoughts?

thanks again.  it's looking good.  now it just needs some decoration.  

also, just a heads-up, I'll be adding a layer (or two) at one point.  i'm working with the Great Schools folks to get high performing schools.

**Comment by stdavis** [comment]
I would say that you can just add commits to this branch if you are comfortable. Otherwise, you could create a new branch off of this one. I'm happy to help you through the git stuff if you need it.

**Comment by gregbunce** [comment]
also, i'll use the post as my starting point for the article and l'll leave the iframe.  super cool.

**Comment by todrobbins** [comment]
Is each hex bin layer supposed to be the same color?

**Comment by gregbunce** [comment]
They start out the same color (empty) but as you alter the sliders (looking at suitability) they begin to render based on your preferences.

**Comment by gregbunce** [comment]
@stdavis, here's the title a brief description/how-to-use/tech details (double-check me on the API links, spatially the OSM one. I used the generic H3 link and not the [js link](https://github.com/uber/h3-js).  it's your call on that one, too.
____________________________________________________________________
(title) 
Neighborhood-level Suitability Analysis in Utah

(description) 
Using GIS, suitability analysis allows a user to spatially explore the appropriateness of an area for the specific criteria that you define. This application highlights areas of suitability on the map based on the weights given in the interactive slider bars. Darker hexagons indicate areas of higher suitability. Conversely, this map can be used to explore areas that are less suitable.

     {show map here}

**Data Sources**
Childcare - [WFRC Child Care Centers GIS layer](https://data.wfrc.org/datasets/utah-child-care-centers)
Hospitals - [WFRC Hospitals GIS layer](https://data.wfrc.org/datasets/utah-hospitals)
Grocery and Food - [WFRC Grocery and Food Stores GIS layer](https://data.wfrc.org/datasets/utah-grocery-and-food-stores-uda)
Public Libraries - [AGRC Public Libraries GIS layer](https://opendata.gis.utah.gov/datasets/utah-public-libraries/)
Charter Schools - [AGRC Schools GIS layer](https://opendata.gis.utah.gov/datasets/utah-schools-prek-to-12)
Colleges and Universities - [AGRC Colleges and Universities GIS layer](https://opendata.gis.utah.gov/datasets/utah-schools-higher-education/)
UTA Commuter Rail Stations - [UTA Commuter Rail Stations GIS layer](https://opendata.gis.utah.gov/datasets/utah-uta-light-rail-stations/)
UTA Bus Stops - [UTA Bus Stops GIS layer](https://opendata.gis.utah.gov/datasets/utah-uta-bus-stops-wasatch-front/)
Community Centers - [WFRC Community Centers GIS layer](https://data.wfrc.org/datasets/community-centers/)
Local Parks - [AGRC Local Parks GIS layer](https://opendata.gis.utah.gov/datasets/utah-parks-local)

**Technical Info**
This map was created using [OSM](https://wiki.openstreetmap.org/wiki/API) and the [Uber H3 Hexagonal Geospatial Indexing System](https://github.com/uber/h3), [resolution 8](https://h3geo.org/docs/core-library/restable).

**Comment by gregbunce** [comment]
also, are you okay if we merge this pr after we're happy with the web map (the HTML page)?  that way matt can pass the map along and it won't be held up on me finishing the blog post.

**Comment by gregbunce** [comment]
two notes worth mentioning on the json sources.

- exactly for two layers, all json sources link to 'live' data via open data/esri hub.
- the reason i had to host two json layers in a gist is b/c
1:  the "parks" layer is a polygon layer and i converted it to centroids for this map (i couldn't get h3.pollyfill to work)
2: the bus stops had null geom in the feature service causing the app to crash.  uta is working on fixing that and then we can point to the live open data json file.

I mentioned this b/c i wonder if you want to put the json for those two layers somewhere other than my github gist.  it's up to you.  I won't delete the files (the json gists), but.. that's your call.

**Comment by gregbunce** [comment]
note: i changed the title of the map to "Neighborhood Suitability Analysis in Utah" (I'll leave the decision up to you for the URL)

okay, i'm done.  i won't make any more changes.  ha!  thanks

**Comment by gregbunce** [comment]
i just noticed that "UTA Commuter Rail Stations" should be changed to "UTA Light Rail Stations"

this correction affects the label on the slider bar and the label in the data sources sections.  the HTML link to the data source is accurate, so that does not need to change - just the hyperlink text.

File: developer/applications/suitability/index.html
Comment (gregbunce) [inline developer/applications/suitability/index.html:150]: the reference to healthdists can be removed.  it's not being used in the sliders or anywhere else.

File: developer/applications/suitability/index.html
Comment (gregbunce) [inline developer/applications/suitability/index.html:136]: this line can be removed. `healthdists` are not being used.

**Review by gregbunce — CHANGES_REQUESTED** [review]

**Comment by stdavis** [comment]
@gregbunce This is ready for another look: https://deploy-preview-1653--gis-utah-gov.netlify.app/developer/applications/suitability

**Review by gregbunce — APPROVED** [review]

**Comment by stdavis** [comment]
This has been deployed. Here's the production URL: https://gis.utah.gov/developer/applications/suitability/

**Pull request merged**
